device_calendar/lib/src/device_calendar.dart

```diff
@@ -74,6 +81,8 @@ class DeviceCalendarPlugin {
       res.errorMessages.add(
           "[${ErrorCodes.invalidArguments}] ${ErrorMessages.invalidMissingCalendarId}");
     }
+
+    // TODO: Extend capability to handle null start or null end (eg all events after a certain date (null end date) or all events prior to a certain date (null start date))
```

Comment:
iOS requires both start and end to be specified if dates are used and this is the only predicate that they allow using
https://developer.apple.com/documentation/eventkit/ekeventstore/1507479-predicateforeventswithstartdate?language=objc

retrieve by id is done using this https://developer.apple.com/documentation/eventkit/ekeventstore/1507490-eventwithidentifier?language=objc

Reply:
Pushing this to the next iteration